projects/sandbox/infer/infer/callback.py

```diff
@@ -21,38 +21,45 @@ class Callback:
     Callable class for handling asynchronous server
     responses for streaming inference across sequences
     of timeseries data. New sequences should be
-    initialized by calling the `start_new_sequence`
-    method before starting to submit requests. Only
-    one sequence can be inferred upon at once.
+    initialized by calling the `initialize` method before
+    starting to submit requests. Only one sequence can be
+    inferred upon at once.
 
     Once inference has completed for each sequence,
     it will be asynchronously convolved with a
-    boxcar filter to perform local integration, then
-    both the raw and integrated timeseries will be
-    written to a file in `write_dir` with the filename
-    `out_{start}-{stop}.hdf5`, where `start` and `stop`
-    indicate the initial and final GPS timestamps of
-    the sequence.
+    boxcar filter to perform local integration, and then
+    the outputs will be clustered to find local maxima to
+    identify as events.
 
     Args:
-        write_dir:
-            Directory to which to save network outputs
-            in HDF5 format.
+        id:
+            Identifier used to match up sequences of
+            inference requests.
         inference_sampling_rate:
             Rate at which to sample inference windows
             from the input timeseries. Represents the
             sample rate of the output timeseries.
         batch_size:
             The number of subsequent windows to
             include in a single batch of inference.
-        window_length:
+        integration_window_length:
             Length of the window over which network
             outputs should be locally integrated,
             specified in seconds.
+        cluster_window_length:
+            Length of the window over which network
+            outputs should be clustered, specified
+            in seconds
+        fduration:
+            Length of the time-domain whitening filter,
+            specified in seconds.
+        psd_length:
+            Length of background to use for PSD calculation,
+            specified in seconds.
     """
 
     id: int
-    sample_rate: float
+    inference_sampling_rate: float
```

Comment:
On the one hand I agree that this helps distinguish what sampling rate this should refer to based on our nomenclature. The reason I opted to call this `sample_rate` in the first place is to emphasize that the output of the NN is just a timeseries with a different sample rate and physical significance. But I think I agree that belaboring that conceptual point is not worth the confusion in terms of code readability

Reply:
Gotcha, I see. Yeah, `sample_rate` comes a few times with slightly different contexts. There's the sample rate of the strain data vs the sample rate of the inference, and there's instances when it determines the sample rate of a time series vs instances where it's just descriptive. Worth thinking about when it comes to standardizing nomenclature.